pyproject.toml

```
@@ -84,23 +84,47 @@ line-length = 99
 
 [tool.ruff.lint]
 explicit-preview-rules = true
-select = ["A", "E", "W", "F", "C", "N", "D", "I", "CPY001"]
+select = [
+  "A",
+  "E",
+  "W",
+  "F",
+  "C",
+  "N",
+  "D",
+  "I",
+  "B",
+  "CPY001",
+  "RUF",
+  "S",
+  "SIM",
+  "UP",
+  "TC",
+]
 ignore = [
-    # Missing docstring in public method (pydocstyle doesn't look for docstrings in super class
-    # https://github.com/PyCQA/pydocstyle/issues/309) TODO: add pylint check? https://github.com/PyCQA/pydocstyle/issues/309#issuecomment-1284142716
-    "D102",
-    "D105", # Missing docstring in magic method
-    "D107", # Missing docstring in __init__
-    "D403", # First word of the first line should be capitalized (false positive on "MySQL")
-    "D415", # Docstring first line punctuation (doesn't make sense for properties)
-    "E501", # Line too long (because using black creates errors with this)
-    "N818", # Exception name should be named with an Error suffix
-    "W505", # Doc line too long (so that strings in comments aren't split across lines)
+  # Missing docstring in public method (pydocstyle doesn't look for docstrings in super class
+  # https://github.com/PyCQA/pydocstyle/issues/309) TODO: add pylint check? https://github.com/PyCQA/pydocstyle/issues/309#issuecomment-1284142716
+  "D102",
+  "D105", # Missing docstring in magic method
+  "D107", # Missing docstring in __init__
+  "D403", # First word of the first line should be capitalized (false positive on "MySQL")
+  "D415", # Docstring first line punctuation (doesn't make sense for properties)
+  "E501", # Line too long (because using black creates errors with this)
+  "N818", # Exception name should be named with an Error suffix
+  "W505", # Doc line too long (so that strings in comments aren't split across lines)
+  "S101",
 ]
 
```

```suggestion

[tool.ruff.lint.ruff]
parenthesize-tuple-in-subscript = true

```
change behavior of ruf031 to be more readable